### ISSUE TITLE
CPU cubic Resize differs from PyTorch bicubic for pytorch_half_pixel with antialias=0

### ISSUE THREAD

**Issue by ALinrunrun** [microsoft/onnxruntime#28292]
### Describe the issue

ONNX Runtime `CPUExecutionProvider` produces different results from PyTorch bicubic interpolation for `Resize` with `mode="cubic"`, `coordinate_transformation_mode="pytorch_half_pixel"`, `cubic_coeff_a=-0.5`, and `antialias=0`.

In the reproducer below, an input tensor of shape `[1, 1, 8, 8]` is resized to `[1, 1, 4, 4]` using scales `[1.0, 1.0, 0.5, 0.5]`.

The result is compared against:

`torch.nn.functional.interpolate(..., mode="bicubic", align_corners=False, antialias=False)`

The maximum absolute difference is `0.06125379`, which is above the `0.05` tolerance used in this test.

### To reproduce

```
import numpy as np
import onnx
from onnx import TensorProto, helper
import onnxruntime as ort
import torch

np.random.seed(11)
src = np.random.rand(1, 1, 8, 8).astype(np.float32)

X = helper.make_tensor_value_info("X", TensorProto.FLOAT, [1, 1, 8, 8])
Y = helper.make_tensor_value_info("Y", TensorProto.FLOAT, [1, 1, 4, 4])
scales = helper.make_tensor(
    "scales",
    TensorProto.FLOAT,
    [4],
    [1.0, 1.0, 0.5, 0.5],
)

node = helper.make_node(
    "Resize",
    inputs=["X", "", "scales"],
    outputs=["Y"],
    mode="cubic",
    coordinate_transformation_mode="pytorch_half_pixel",
    cubic_coeff_a=-0.5,
    antialias=0,
)

graph = helper.make_graph([node], "cubic_resize", [X], [Y], initializer=[scales])
model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 18)])
model.ir_version = 9
onnx.checker.check_model(model)

sess = ort.InferenceSession(
    model.SerializeToString(),
    providers=["CPUExecutionProvider"],
)

ort_out = sess.run(None, {"X": src})[0]

torch_ref = (
    torch.nn.functional.interpolate(
        torch.from_numpy(src),
        size=(4, 4),
        mode="bicubic",
        align_corners=False,
        antialias=False,
    )
    .numpy()
)

max_abs_diff = float(np.max(np.abs(ort_out - torch_ref)))

print("ORT CPU output:")
print(ort_out[0, 0])
print("PyTorch output:")
print(torch_ref[0, 0])
print("max_abs_diff:", max_abs_diff)
print("PASS=", max_abs_diff <= 0.05)
```

### Expected output
```
max_abs_diff <= 0.05
PASS=True
```
### Actual output
```
ORT CPU output:
[[ 0.54097813  0.51120776  0.7225869   0.29133016]
 [-0.03344578  0.46032864  0.5131942   0.464438  ]
 [ 0.45801762  0.45003796  0.42054695  0.71767175]
 [ 0.57702196  0.5061369   0.57921034  0.3523327 ]]
PyTorch output:
[[ 0.56336266  0.5119769   0.75297356  0.27415875]
 [-0.09469957  0.47234362  0.50275916  0.4582208 ]
 [ 0.47150892  0.4341494   0.42115048  0.73556787]
 [ 0.596296    0.512735    0.5854972   0.33359376]]
max_abs_diff: 0.06125379353761673
PASS=False
```

### Platform

Linux

### OS Version

Linux-6.17.0-20-generic-x86_64-with-glibc2.39

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.25.1

### ONNX Runtime API

Python

### Architecture

X86

### Execution Provider

Default CPU

**Comment by yuslepukhin** [comment]
The discrepancy is a spec difference between ONNX and PyTorch.
The likely source of the 0.06 max diff. For cubic interpolation, 4 pixels are sampled (floor(x)−1 to floor(x)+2). At boundaries, ORT clamps indices to [0, len−1] per the ONNX spec. PyTorch uses its own boundary padding logic which may differ subtly. With downscaling (8→4) and x_original = 0.5, ORT samples index −1 (clamped to 0), while PyTorch may handle this differently.

**Comment by beanduan22** [comment]
Thanks for the explanation. That makes sense.

I just wanted to clarify one thing from a user perspective: since the model uses `coordinate_transformation_mode="pytorch_half_pixel"` and compares against PyTorch `interpolate(..., mode="bicubic", align_corners=False, antialias=False)`, I initially expected the result to be close to PyTorch behavior.

If ORT is following the ONNX spec correctly but PyTorch uses different boundary handling for bicubic interpolation, would it be possible to add a short documentation note or test case to make this difference explicit? This could help avoid confusion for users expecting `pytorch_half_pixel` to imply closer PyTorch compatibility.

Thanks again for looking into this.

**Comment by yuslepukhin** [comment]
See the PR attached to your other issue.
